app/components/projects/status_button_component.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class Projects::StatusButtonComponent < ApplicationComponent
+  include ApplicationHelper
+  include OpTurbo::Streamable
+  include OpPrimer::ComponentHelpers
+  include ProjectStatusHelper
+
+  attr_reader :project, :user
+
+  def initialize(project:, user:, size: :medium)
+    super
+
+    @project = project
+    @user = user
+    @size = size
+
+    @status = find_status(project.status_code)
+  end
+
+  def call
+    component_wrapper do
+      render(
+        OpPrimer::StatusButtonComponent.new(
+          current_status: build_item(@status),
+          items: build_items,
+          readonly: !edit_enabled?,
+          disabled: !edit_enabled?,
+          button_arguments: {
+            title: I18n.t("js.label_edit_project_status"),
+            size: @size
+          },
+          menu_arguments: {
+            select_variant: :single
+          }
```

Comment:
I think you can remove that and fallback to the default selection strategy which is `:none`. Then you can get rid of the additional CSS file and class you introduced.